File: py/1861.py
def main():
    frequencies = dict()
    dead = set()

    while True:
        try:
            murderer, murdered = input().split()
            dead.add(murdered)

            if murderer not in frequencies:
                frequencies[murderer] = 0

            frequencies[murderer] += 1

        except EOFError:
            break

    for i in dead:
        try:
            frequencies.pop(i)
        except KeyError:
            continue

    print('HALL OF MURDERERS')
    murderers = sorted(frequencies)

    for i in murderers:
        print(i, frequencies[i])


main()

File: cpp/1861.cpp
#include <iostream>
#include <string>
#include <map>
#include <queue>

using namespace std;

int main()
{
    string murderer, murdered;
    map<string, int> frequencies;
    queue<string> dead;

    while (cin >> murderer >> murdered) {
        dead.push(murdered);
        frequencies[murderer]++;
    }

    while (!dead.empty()) {
        frequencies.erase(dead.front());
        dead.pop();
    }

    cout << "HALL OF MURDERERS\n";

    for (auto i: frequencies) {
        cout << i.first << ' ' << i.second << '\n';
    }

    return 0;
}